lib/vm/src/cache.rs

```diff
@@ -73,7 +73,12 @@ where
     }
 
     /// get instance returns a wasmer Instance tied to a previously saved wasm
-    pub fn get_instance(&mut self, id: &[u8], deps: Extern<S, A>) -> Result<Instance<S, A>, Error> {
+    pub fn get_instance(
+        &mut self,
+        id: &[u8],
+        deps: Extern<S, A>,
+        gas_limit: u64,
+    ) -> Result<Instance<S, A>, Error> {
         let hash = WasmHash::generate(&id);
 
         // pop from lru cache if present
```

Comment:
The issue with this PR, is that if I pull an Instance off the lru cache, it will have the same gas limit as the first time.

Try this, by running some code (with lru), get_instance the first time with a too low gas, and it fails. Then put it back on cache. Then try `get_instance` with more than enough gas. I believe it will fail again as well.

Reply:
I am working on a dedicated test for the recovering after out of gas now.

Reply:
Done in 201829cf84a6f0a6608ac25d4ad334731726a04f